BL0937 accuracy/res improvement by slow down readout

listen-obk opened the pull request:
adopt measure intervals to output frequency of bl0937 for more resolution  (~1,7W at 1Hz)

listen-obk committed:
Update pre_build.sh

listen-obk committed:
slower interval drv_bl0937.c

frequency > Hz for low power

listen-obk committed:
fix warning, refactor pulse handling and timestamp calculations

Updated pulse handling to account for overflow and changed timestamp calculations.

listen-obk committed:
add command Interval

Added command to set measurement interval dynamically.

listen-obk committed:
Refactor pulse calculation logic in drv_bl0937.c

dynamic interval

listen-obk committed:
Add CMake workflow for multi-platform builds

listen-obk committed:
fix tick calc

listen-obk committed:
Delete multi-platform CMake workflow

Removed the multi-platform CMake workflow configuration.

listen-obk committed:
Adjust threshold and log message formatting

listen-obk committed:
Add debug logging for scaled voltage, current, and power

listen-obk committed:
apply volt scale factor to power

Added calibration variables for voltage and power measurements.

listen-obk committed:
Update calibration constants and measurement intervals v c

listen-obk committed:
Refactor conditionals and logging in drv_bl0937.c

listen-obk committed:
Refactor pulse handling logic in drv_bl0937.c

listen-obk committed:
Update pulse counting and measurement logic

Refactor pulse counting and measurement intervals for BL0937 driver.

listen-obk committed:
Refactor frequency calculations and logging

listen-obk committed:
Refactor pulse timestamp and measurement logic

Refactor pulse timestamp handling and measurement calculations for voltage, current, and power. Introduce debug logging for pulse timestamps and measurement changes.

divadiow commented:
This looks interesting. What's the effect of OTA updating to this version of you've already got BL0937 configured? Does it breaks stats or keep working as before? 

Have you tried it on multiple platforms?

listen-obk committed:
add min/max interval functionality and command

listen-obk commented:
Currently I test only on new Tuya plugs with EC6600. OTA from .205 (and to .207) main version worked well but new calibration is required. Because I wanted to see frequency of BL0937 in log and due to restrictions of Scale (raw values in int) I added a factor of 100 for V and 1000 for I and P. There might be better solutions for that.

I did not check impact on stats yet - I would expect it will work well.

listen-obk committed:
add command for scalefactor migration

listen-obk committed:
fix timestamp overflow 

32bit only

listen-obk committed:
Merge branch 'openshwprojects:main' into main

listen-obk committed:
change min of new cal value

listen-obk committed:
change cal for current (no factor)

listen-obk committed:
Merge branch 'openshwprojects:main' into main

listen-obk committed:
change response to command interval

listen-obk committed:
Remove change

listen-obk committed:
add power roc recognition, change defaults to match current main branch

listen-obk committed:
Update work directory gh_prebuild.sh

listen-obk committed:
Merge branch 'openshwprojects:main' into main

listen-obk committed:
change debug message

listen-obk committed:
Merge branch 'openshwprojects:main' into main

listen-obk committed:
add int BL0937_utlGetFactor10

listen-obk committed:
test scale/multiply

listen-obk committed:
Update drv_bl0937.c

listen-obk committed:
Update drv_bl0937.c

listen-obk committed:
Merge branch 'openshwprojects:main' into main

listen-obk committed:
fix scale multiply, change messages

listen-obk committed:
function rename

listen-obk committed:
Merge branch 'openshwprojects:main' into main

listen-obk committed:
add check NTP to local uptime sec

Temporary to check for accuracy of frequency calculation

listen-obk committed:
Update drv_pwrCal.c

listen-obk committed:
Add MQTT message on commands

listen-obk committed:
Add MQTT message to commands

listen-obk committed:
fix datatype 

general uint not for simulator platform

listen-obk committed:
fix datatype

listen-obk committed:
Update drv_bl0937.c

listen-obk committed:
reduce compiler warnings

listen-obk committed:
Update new_mqtt.c

listen-obk committed:
fix factor recognition

listen-obk committed:
add command sendtimestamps

listen-obk committed:
Merge branch 'openshwprojects:main' into main

listen-obk committed:
correct wrong commit

listen-obk committed:
add command sendtimestamp

listen-obk committed:
fix

listen-obk committed:
fx

listen-obk committed:
correct cmd desc

listen-obk committed:
fix forceonroc

listen-obk committed:
remove limit

listen-obk committed:
make reboot indicator visible

listen-obk committed:
make reboot indicator visible

listen-obk committed:
Refactor sample time/count limits and logging

Updated security limits for sample time and count, and enhanced logging for consumption history.

listen-obk committed:
update on restart, improve command/logging

listen-obk committed:
Merge branch 'openshwprojects:main' into main

listen-obk committed:
fix build err, add offset for ntp diff

listen-obk committed:
Merge branch 'openshwprojects:main' into main

listen-obk committed:
fix timestamp check

listen-obk committed:
number format

listen-obk committed:
number format

listen-obk committed:
Merge branch 'openshwprojects:main' into main

listen-obk committed:
Add diff_ntp_count

listen-obk committed:
fix forceonpwr for low settings

Updated precision handling and logging for power calculations.

listen-obk committed:
Merge branch 'openshwprojects:main' into main
